### PR TITLE
Adapted to kube-rs 0.60 and operator-rs 0.2.2

### PR DESCRIPTION
**Pull request by maltesander**
## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: src/test/kube.rs [hunk 1]
```diff
@@ -414,7 +414,10 @@ impl KubeClient {
         K: Clone + Debug + DeserializeOwned + Resource,
         <K as Resource>::DynamicType: Default,
     {
-        let get_value = |resource: &K| resource.meta().annotations.get(key).cloned();
+        let get_value = |resource: &K| match &resource.meta().annotations {
+            Some(annotations) => annotations.get(key).cloned(),
+            _ => None,
+        };
```
Comment (siegfriedweber) [inline src/test/kube.rs:420]: This is okay. Personally, I find a chain more readable than a match:

```suggestion
        let get_value = |resource: &K| {
            resource
                .meta()
                .annotations
                .as_ref()
                .and_then(|annotations| annotations.get(key).cloned())
        };
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: src/operator/setup.rs [hunk 2]
```diff
@@ -88,26 +88,25 @@ where
...
+                } else {
+                    return Err(anyhow!(
                 "Pod [{}] has no version label [{}]. Expected version [{}]. This should not happen!",
                 pod.metadata.name.as_ref().unwrap(),
                 stackable_operator::labels::APP_VERSION_LABEL,
                 version.to_string(),
             ));
+                }
             }
         }
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/operator/setup.rs:111]: Oh yeah thanks.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -2,6 +2,14 @@
 
 == 0.5.0 - unreleased
 
+:25: https://github.com/stackabletech/integration-test-commons/pull/25[#25]
+
+### Changed
+
+- `kube-rs`: `0.58` → `0.60` ({25}).
+- `k8s-openapi` features: `v1_21` → `v1_22` ({25}).
```
[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc:10]: Yeah i messed up some copy paste here.

File: src/test/kube.rs (hunk 1 above)
Comment (maltesander) [reply src/test/kube.rs:420]: done

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**